contracts/implementation/PoolCommitter.sol

```diff
@@ -59,301 +47,251 @@ contract PoolCommitter is IPoolCommitter, Ownable {
-     * @param direction UP if going from earliest to latest, DOWN if going from latest to earliest.
+     * @notice Claim user's balance. This can be done either by the user themself or by somebody else on their behalf.
      */
-    function skipDeletedMiddleCommits(ScanDirection direction) internal {
-        if (direction == ScanDirection.UP) {
-            uint128 nextEarliestCommitUnexecuted = earliestCommitUnexecuted;
-            while (nextEarliestCommitUnexecuted <= latestCommitUnexecuted) {
-                IPoolCommitter.Commit memory _commit = commits[nextEarliestCommitUnexecuted];
-                if (_commit.owner == address(0)) {
-                    // Commit deleted (uncommitted) or already executed
-                    nextEarliestCommitUnexecuted += 1; // It makes sense to set the next unexecuted to the next number
-                    continue;
-                } else {
-                    break;
-                }
-            }
-            if (nextEarliestCommitUnexecuted > latestCommitUnexecuted) {
-                // We have just bumped earliestCommitUnexecuted above latestCommitUnexecuted,
-                // we have therefore run out of commits
-                earliestCommitUnexecuted = NO_COMMITS_REMAINING;
-            } else {
-                earliestCommitUnexecuted = nextEarliestCommitUnexecuted;
-            }
+    function claim(address user) external override updateBalance {
+        Balance memory balance = userAggregateBalance[user];
+        ILeveragedPool pool = ILeveragedPool(leveragedPool);
+        if (balance.settlementTokens > 0) {
+            pool.quoteTokenTransfer(user, balance.settlementTokens);
         }
-
-        if (direction == ScanDirection.DOWN) {
-            uint128 nextLatestCommitUnexecuted = latestCommitUnexecuted;
-            while (nextLatestCommitUnexecuted >= earliestCommitUnexecuted) {
-                IPoolCommitter.Commit memory _commit = commits[nextLatestCommitUnexecuted];
-                if (_commit.owner == address(0)) {
-                    // Commit deleted (uncommitted) or already executed
-                    nextLatestCommitUnexecuted -= 1;
-                    continue;
-                } else {
-                    break;
-                }
-            }
-            if (nextLatestCommitUnexecuted < earliestCommitUnexecuted) {
-                // We have just bumped earliestCommitUnexecuted above latestCommitUnexecuted,
-                // we have therefore run out of commits
-                earliestCommitUnexecuted = NO_COMMITS_REMAINING;
-            } else {
-                latestCommitUnexecuted = nextLatestCommitUnexecuted;
-            }
+        if (balance.longTokens > 0) {
+            pool.poolTokenTransfer(LONG_INDEX, user, balance.longTokens);
+        }
+        if (balance.shortTokens > 0) {
+            pool.poolTokenTransfer(SHORT_INDEX, user, balance.shortTokens);
         }
+        delete userAggregateBalance[user];
```

Comment:
whats difference between this and just setting it to 0?

Reply:
`delete` also refunds some gas back to the caller: https://ethereum.stackexchange.com/a/58496
@Paul4912